app/javascript/tests/patient/laboratory/FirstPositiveLaboratory.test.js

```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+import { Button } from 'react-bootstrap';
+import FirstPositiveLaboratory from '../../../components/patient/laboratory/FirstPositiveLaboratory';
+import { mockLaboratory1 } from '../../mocks/mockLaboratories';
+
+import LaboratoryModal from '../../../components/patient/laboratory/LaboratoryModal';
+
+function getWrapper(lab, onChange) {
+  return shallow(<FirstPositiveLaboratory lab={lab} onChange={onChange} />);
+}
+
+describe('FirstPositiveLaboratory', () => {
+  it('Properly renders a button to enter a lab result if lab is null', () => {
+    const wrapper = getWrapper(null);
+    wrapper.find(Button).simulate('click');
+    expect(wrapper.find(LaboratoryModal).exists()).toBeTruthy();
+  });
+
+  it('Properly renders lab result details when lab is present', () => {
+    const wrapper = getWrapper(mockLaboratory1);
+    expect(wrapper.find('#edit-first_positive_lab').exists()).toBeTruthy();
+    expect(wrapper.find('#delete-first_positive_lab').exists()).toBeTruthy();
+    expect(wrapper.find('.first-positive-lab-result-field-name').exists()).toBeTruthy();
+  });
+
+  it('Properly opens LaboratoryModal when the edit button is clicked', () => {
+    const wrapper = getWrapper(mockLaboratory1);
+    wrapper.find('#edit-first_positive_lab').simulate('click');
+    expect(wrapper.find(LaboratoryModal).exists()).toBeTruthy();
+  });
+
+  it('Properly calls onChange with a null lab when the delete button is clicked', () => {
+    const onChange = jest.fn();
```

Comment:
can we move this declaration above and just always put the const as the prop? (we do this elsewhere)

Reply:
this is no longer relevant since it's been replaced by a different spy function: https://github.com/SaraAlert/SaraAlert/pull/861/commits/a5f7d0708f903826cb1f4dec5169309b37c77dbb#diff-30c5340a7ce13eb30d2abf50df7d2d5821ef5e3ff48852f2a949a137fbc55774R55